src/Integrations/Integrations/Logs/LogsIntegration.php

```diff
@@ -0,0 +1,265 @@
+
+class LogsIntegration extends Integration
+{
+    const NAME = 'logs';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return self::NAME;
+    }
+
+    public static function laminasLogLevelToString(int $logLevel, string $fallback): string
+    {
+        switch ($logLevel) {
+            case 0:
+                return 'emergency';
+            case 1:
+                return 'alert';
+            case 2:
+                return 'critical';
+            case 3:
+                return 'error';
+            case 4:
+                return 'warning';
+            case 5:
+                return 'notice';
+            case 6:
+                return 'info';
+            case 7:
+                return 'debug';
+            default:
+                return $fallback;
+        }
+    }
+
+    public static function getPlaceholders(
+        string $levelName,
+        string $traceIdSubstitute = null,
+        string $spanIdSubstitute = null
+    ): array {
+        $placeholders = [
+            '%dd.trace_id%' => 'dd.trace_id="' . ($traceIdSubstitute ?? logs_correlation_trace_id()) . '"',
+            '%dd.span_id%'  => 'dd.span_id="' . ($spanIdSubstitute ?? dd_trace_peek_span_id()) . '"',
+        ];
+
+        $appName = ddtrace_config_app_name();
+        if ($appName) {
+            $placeholders['%dd.service%'] = 'dd.service="' . $appName . '"';
+        } else {
+            $placeholders['%dd.service%'] = '';
+        }
+
+        $currentContext = \DDTrace\current_context();
+        if ($currentContext['version']) {
+            $placeholders['%dd.version%'] = 'dd.version="' . $currentContext['version'] . '"';
+        } else {
+            $placeholders['%dd.version%'] = '';
+        }
+
+        if ($currentContext['env']) {
+            $placeholders['%dd.env%'] = 'dd.env="' . $currentContext['env'] . '"';
+        } else {
+            $placeholders['%dd.env%'] = '';
+        }
+
+        $placeholders['%status%'] = 'status="' . $levelName . '"';
+
+        return $placeholders;
+    }
+
+    public static function messageContainsPlaceholders(string $message, string $levelName): bool
+    {
+        $placeholders = LogsIntegration::getPlaceholders($levelName);
+
+        foreach ($placeholders as $placeholder => $value) {
+            if (str_contains($message, $placeholder)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public static function appendTraceIdentifiersToMessage(
+        string $message,
+        string $levelName,
+        string $traceIdSubstitute = null,
+        string $spanIdSubstitute = null
+    ): string {
+        if (strpos($message, 'dd.trace_id=') !== false) {
+            // Trace identifiers were already appended in the call stack
+            return $message;
+        }
+
+        $additional = "";
+
+        $placeholders = LogsIntegration::getPlaceholders($levelName, $traceIdSubstitute, $spanIdSubstitute);
```

Comment:
Seems useless as this function is called from a elseif where we know we don't have placeholders